app/src/lib/stores/updates/update-remote-url.ts

```diff
@@ -0,0 +1,29 @@
+import { Repository } from '../../../models/repository'
+import { IAPIRepository } from '../../api'
+import { GitStore } from '../git-store'
+
+export const updateRemoteUrl = async (
+  gitStore: GitStore,
+  repository: Repository,
+  apiRepo: IAPIRepository
+): Promise<void> => {
+  // I'm not sure when these early exit conditions would be met. But when they are
+  // we don't have enough information to continue so exit early!
+  if (!gitStore.currentRemote) {
+    return
+  }
+  if (!repository.gitHubRepository) {
+    return
+  }
+
+  const remoteUrl = gitStore.currentRemote.url
+  const updatedRemoteUrl = apiRepo.clone_url
+  const isProtocolHttps = remoteUrl && remoteUrl.startsWith('https://')
```

Comment:
This is checking that the current remote protocol is `https`. What I was concerned about in https://github.com/desktop/desktop/pull/8132#discussion_r316675400 was mainly that we'd be switching protocols from https to ssh given our poor support of ssh. I think perhaps a safer logic here would be to check if the protocol of the current remote url matches that of the new clone url and to only update it if they do.

```ts
protocolEquals = url.parse(remoteUrl).protocol === url.parse(updatedRemoteUrl).protocol
```